guides/common/modules/proc_salt-authenticating-salt-minions-using-host-names.adoc

```diff
@@ -24,5 +30,5 @@ permissive_pki_access: True
 ----
 # touch /etc/salt/autosign.conf
 # chgrp foreman-proxy /etc/salt/autosign.conf
-# chmod 660 /etc/salt/autosign.conf
+# chmod 640 /etc/salt/autosign.conf
```

Comment:
@bochi Can you please doublecheck that these permissions are fine? I'm not an expert to review this factually :) Thank you!

Reply:
I was looking at the installer, but it surprises me we don't ensure the file exists with the correct permissions.

These permissions also look a bit odd. The result is that it will be owned by `root` and the group is `foreman-proxy`. It is then only writable for `root` and `foreman-proxy` can read it. Is that correct? It implies there is either no automation to modify `autosign.conf` or there's a process running as `root` that does.

Reply:
@sbernhard @bochi Please, can you help us to move this forward?

Reply:
@bochi wrote in https://github.com/theforeman/foreman_salt/pull/136#discussion_r627233869
> I actually have exactly that setup working on salt 3003 :-)
The only thing that needs to be confiured is that the user running salt needs to be in the foreman-proxy group and the configuration option permissive_pki_access: True in the salt master config needs to be set.

Means, the following works:

In /etc/salt/master the configuration need to look like this:
`autosign_file: /etc/salt/autosign.conf
permissive_pki_access: True`

The file permission (user, group) of /etc/salt/autosign.conf  need to look like this:
`-rw-rw----. 1 root foreman-proxy 37 10. Mai 22:24 /etc/salt/autosign.conf`

In /etc/group the user who runs salt (in this case 'root') need to be in the group:
`foreman-proxy:x:993:root`